databricks_cli/pipelines/cli.py

```diff
@@ -0,0 +1,121 @@
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deploys a delta pipeline')
+@click.argument('spec_arg', default=None, required=False)
+@click.option('--spec', default=None, help=PipelinesSpecPathClickType.help)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def deploy_cli(api_client, spec_arg, spec):
+    """
+    deploys a pipeline provisioned by the spec at the location provided
+    :param api_client:
+    :param spec_arg:
+    :param spec:
+    :return:
+    """
+    if not bool(spec_arg) ^ bool(spec):
+        raise RuntimeError('The spec should be provided either by an option or argument')
+    src = spec_arg if bool(spec_arg) else spec
+    spec_obj = _read_spec(src)
+    PipelinesApi(api_client).deploy(spec_obj)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deletes a delta pipeline')
+@click.argument('spec_arg', default=None, required=False)
+@click.option('--spec', default=None, help=PipelinesSpecPathClickType.help)
+@click.option('--pipeline-id', default=None, help='id associated with pipeline')
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_cli(api_client, spec_arg, spec, pipeline_id):
+    """
+    deletes a pipeline provisioned by the spec at the location provided or id
+    :param api_client:
+    :param spec:
+    :param pipeline_id:
+    :return:
+    """
+    # Only one out of spec/pipeline_id/spec_arg should be supplied
+    if not ((not (bool(spec_arg) and bool(spec) and bool(pipeline_id)))
+            and (bool(spec_arg) ^ bool(spec) ^ bool(pipeline_id))):
+        raise RuntimeError('Either spec should be provided as an argument '
+                           'or option, or the pipeline-id should be provided')
+    if bool(spec_arg) ^ bool(spec):
+        src = spec_arg if bool(spec_arg) else spec
+        pipeline_id = _read_spec(src)["id"]
+    PipelinesApi(api_client).delete(pipeline_id)
+
+
+def _read_spec(src):
+    """
+    Reads the spec at src as a JSON if no file extension is provided, or if in the extension format
+    if the format is supported.
+    :param src:
+    :return: json of the spec
+    """
+    extension = os.path.splitext(src)[1]
+    if extension == '' or extension.lower() == '.json':
+        return _read_json_spec(src)
+    else:
+        raise RuntimeError('The provided file extension for the spec is not supported')
+
+
+def _read_json_spec(src):
+    with open(src, 'r') as f:
+        json = f.read()
+    return json_loads(json)
+
+
+@click.group(context_settings=CONTEXT_SETTINGS,
+             short_help='Utility to interact with the Databricks Delta Pipelines.')
+@click.option('--version', '-v', is_flag=True, callback=print_version_callback,
+              expose_value=False, is_eager=True, help=version)
+@debug_option
+@profile_option
+def pipelines_group():
+    """
+    Utility to interact with the Databricks pipelines.
+    """
+    pass
```

Comment:
This is used to add pipelines so that you can say databricks pipelines ... 

pipelines group is added to the databricks command, and deploy/delete are added to pipelines group. By itself, the function doesn't do much. The short help string is printed if someone does databricks pipelines -h